api/v1alpha1/externaldns_types.go

```diff
@@ -18,41 +18,412 @@ package v1alpha1
+const (
+	ServicePublishPolicyExternalOnly        ServicePublishPolicy = "PublishExternal"
+	ServicePublishPolicyExternalAndInternal ServicePublishPolicy = "PublishInternalExternal"
+)
+
+type ExternalDNSCRDSourceOptions struct {
+	// Kind is the kind of the CRD
+	// source resource type to be
+	// consumed by ExternalDNS.
+	//
+	// e.g. "DNSEndpoint"
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Kind string `json:"kind"`
+
+	// Version is the API version
+	// of the given resource kind for
+	// ExternalDNS to use.
+	//
+	// e.g. "externaldns.k8s.io/v1alpha1"
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Version string `json:"version"`
+
+	// LabelFilter specifies a label filter
+	// to be used to filter CRD resource instances.
+	// Only one label filter can be specified on
+	// an ExternalDNS instance.
+	//
+	// +optional
+	LabelFilter map[string]string `json:"labelFilter,omitempty"`
+}
+
+// ExternalDNSStatus defines the observed state of ExternalDNS.
 type ExternalDNSStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Conditions is a list of operator-specific conditions
+	// and their status.
+	Conditions []ExternalDNSOperatorCondition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the most recent generation observed.
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// Provider is the configured provider in use by ExternalDNS.
+	Provider ExternalDNSProvider `json:"provider,omitempty"`
+
+	// Zones is the configured zones in use by ExternalDNS.
+	Zones []string `json:"zones,omitempty"`
 }
 
-//+kubebuilder:object:root=true
-//+kubebuilder:subresource:status
+type ExternalDNSOperatorCondition struct {
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Type string `json:"type"`
 
-// ExternalDNS is the Schema for the externaldns API
-type ExternalDNS struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Status string `json:"status"`
 
-	Spec   ExternalDNSSpec   `json:"spec,omitempty"`
-	Status ExternalDNSStatus `json:"status,omitempty"`
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+
+	// +optional
+	Reason *string `json:"reason,omitempty"`
+
+	// +optional
+	Message *string `json:"message,omitempty"`
 }
 
-//+kubebuilder:object:root=true
+var (
+	// Available indicates that the ExternalDNS is available.
+	ExternalDNSAvailableConditionType = "Available"
+
+	// AuthenticationFailed indicates that there were issues starting
+	// ExternalDNS pods related to the given provider credentials.
+	ExternalDNSProviderAuthFailedReasonType = "AuthenticationFailed"
```

Comment:
Change the world.  you can use this if there is some external force on you, but also use Available and Degraded, where degraded could have AuthenticationFailed as a reason.

Reply:
this variable is for a reason type, maybe im missing something?

Reply:
> this variable is for a reason type, maybe im missing something?

Nope, I didn't read the block carefully enough.